homeassistant/components/elmax/config_flow.py

```diff
@@ -170,25 +177,29 @@ async def async_step_panels(
 
     async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
         """Perform reauth upon an API authentication error."""
-        self._reauth_username = entry_data.get(CONF_ELMAX_USERNAME)
         self._reauth_panelid = entry_data.get(CONF_ELMAX_PANEL_ID)
         return await self.async_step_reauth_confirm()
 
-    async def async_step_reauth_confirm(self, user_input=None):
+    async def async_step_reauth_confirm(self, user_input=None) -> FlowResult:
         """Handle reauthorization flow."""
         errors = {}
         if user_input is not None:
-            panel_pin = user_input.get(CONF_ELMAX_PANEL_PIN)
+            username = user_input.get(CONF_ELMAX_USERNAME)
             password = user_input.get(CONF_ELMAX_PASSWORD)
+            panel_pin = user_input.get(CONF_ELMAX_PANEL_PIN)
             entry = await self.async_set_unique_id(self._reauth_panelid)
 
+            # This is an edge case: if re-auth flow is triggered, it means that the integration is already
+            # configured, therefore the async_set_unique_id should return the entry. If this does not happen,
+            # just start over the configuration from user step.
+            if entry is None:
+                return await self.async_step_user()
+
```

Comment:
How would this be possible?

Reply:
Like I said, that's an edge case. I actually tried to trigger this case by having two tabs, triggering the reauth in one tab and unconfigure the integration from the other tab, then continue the flow on the first tab.
Luckily, it seems HA traps the "invalid flow specified" before the invocation of this step. Thus, you're right, this is not necessary and I've been happy to remove that piece of logic.